src/libpmem2/badblocks.h

```diff
@@ -26,6 +28,39 @@ extern "C" {
 #define BB_NOT_SUPP \
 	"checking bad blocks is not supported on this OS, please switch off the CHECK_BAD_BLOCKS compat feature using 'pmempool-feature'"
 
+struct pmem2_badblock_context {
```

Comment:
@pbalcer  Yes, it has to be in a header file, so that implementing unit tests for `pmem2_badblock_*()` API was possible, but it can be in a separate header file, of course.

I moved `pmem2_badblock_context` to a separate header file `pmem2_badblock_context.h`